golang/src/tcpcall/server.go

```diff
@@ -26,8 +26,8 @@ type Server struct {
 	bindAddr    string
 	config      ServerConf
 	socket      net.Listener
+	lock        sync.RWMutex
```

Comment:
Agree that it may not be that obvious, but there will be no copying afterwards. In documentation it states that mutex should not be copied(should be understood as created new mutex and set it to "lock" value).
But as you can see in documentation that you gave there is no value recievers on mutexes, only pointer, and as stated here: https://golang.org/ref/spec#Calls : "If x is addressable and &x's method set contains m, x.m() is shorthand for (&x).m()"
So "you MUST use pointer to the same mutex" - no, you may not, if calling it methods. This is valid and will not violate any Golang spec rules.

Reply:
Ok, I'll learn more about it then.